huaweicloud/services/rds/data_source_huaweicloud_rds_flavors.go

```diff
@@ -0,0 +1,205 @@
+							Computed: true,
+						},
+						"availability_zones": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+						"db_versions": {
+							Type:     schema.TypeList,
+							Computed: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceRdsFlavorRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	client, err := config.RdsV3Client(config.GetRegion(d))
+	if err != nil {
+		return fmtp.DiagErrorf("Error creating HuaweiCloud rds client: %s", err)
+	}
+
+	dbType := d.Get("db_type").(string)
+	listOpts := flavors.DbFlavorsOpts{Versionname: d.Get("db_version").(string)}
+
+	pages, err := flavors.List(client, listOpts, dbType).AllPages()
+	if err != nil {
+		return fmtp.DiagErrorf(err.Error())
+	}
+
+	flavorsResp, err := flavors.ExtractDbFlavors(pages)
+	if err != nil {
+		return fmtp.DiagErrorf("Unable to retrieve RDS flavors: %s", err)
+	}
+
+	mode := d.Get("instance_mode").(string)
+	mem := d.Get("memory").(int)
+	az := d.Get("availability_zone").(string)
+	version := d.Get("db_version").(string)
+
+	var vcpus string
+	if v, ok := d.GetOk("vcpus"); ok {
+		vcpus = strconv.Itoa(v.(int))
+	}
+
+	filter := map[string]interface{}{
+		"Vcpus":        vcpus,
+		"Ram":          mem,
+		"Instancemode": mode,
+	}
+	if mem, ok := d.GetOk("memory"); ok {
+		filter["Ram"] = mem.(int)
+	}
+
+	filterFlavors, err := utils.FilterSliceWithField(flavorsResp.Flavorslist, filter)
+
+	if err != nil {
+		return fmtp.DiagErrorf("filter RDS flavors failed: %s", err)
+	}
+
+	var resultFlavors []interface{}
+	var ids []string
+
+	for _, item := range filterFlavors {
+		flavor := item.(flavors.Flavors)
+
+		// filter availability_zones
+		var azList []string
+		for k, v := range flavor.Azstatus {
+			if v == "normal" && (az == "" || (az != "" && az == k)) {
+				azList = append(azList, k)
+			}
+		}
+
+		// filter db_versions
+		var versionList []string
+		for _, v := range flavor.VersionName {
+			if version == "" || (version != "" && version == v) {
+				versionList = append(versionList, v)
+			}
+		}
+
+		if len(azList) > 0 && len(versionList) > 0 {
+			resultFlavors = append(resultFlavors, flattenRdsFlavor(flavor, azList, versionList))
+		}
+
+		ids = append(ids, flavor.ID)
+
+	}
+
+	logp.Printf("[DEBUG]RDS flavors api return:%d, after filter: %d, %v", len(flavorsResp.Flavorslist), len(resultFlavors), resultFlavors)
+
+	mErr := d.Set("flavors", resultFlavors)
+	if mErr != nil {
+		return fmtp.DiagErrorf("set flavors err:%s", mErr)
+	}
+
+	d.SetId(hashcode.Strings(ids))
+
+	return nil
+}
+
+func flattenRdsFlavor(flavor flavors.Flavors, azList, versionList []string) map[string]interface{} {
+	vcpus, _ := strconv.Atoi(flavor.Vcpus)
+
+	return map[string]interface{}{
+		"id":                 flavor.ID,
+		"name":               flavor.Speccode,
```

Comment:
is `name` equal to `id` ?

Reply:
no, 
id is uuid
name like this: `rds.mysql.x1.4xlarge.2.rr`